core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/AnalysisTempFolderProvider.java

```diff
@@ -19,61 +19,47 @@
  */
 package org.sonarsource.sonarlint.core.container.analysis;
 
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import org.picocontainer.ComponentLifecycle;
-import org.picocontainer.PicoContainer;
+import java.io.File;
 import org.picocontainer.injectors.ProviderAdapter;
 import org.sonar.api.utils.TempFolder;
-import org.sonar.api.utils.internal.DefaultTempFolder;
-import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneAnalysisConfiguration;
 
-public class AnalysisTempFolderProvider extends ProviderAdapter implements ComponentLifecycle<TempFolder> {
-  static final String TMP_NAME = ".sonartmp";
-  private DefaultTempFolder projectTempFolder;
-  private boolean started = false;
+public class AnalysisTempFolderProvider extends ProviderAdapter {
 
-  public TempFolder provide(StandaloneAnalysisConfiguration analysisConfig) {
-    if (projectTempFolder == null) {
-      Path workingDir = analysisConfig.workDir();
-      Path tempDir = workingDir.normalize().resolve(TMP_NAME);
-      try {
-        Files.deleteIfExists(tempDir);
-        Files.createDirectories(tempDir);
-      } catch (IOException e) {
-        throw new IllegalStateException("Unable to create root temp directory " + tempDir, e);
-      }
+  private final NoTempFilesDuringAnalysis instance = new NoTempFilesDuringAnalysis();
 
-      projectTempFolder = new DefaultTempFolder(tempDir.toFile(), true);
-    }
-    return projectTempFolder;
+  public TempFolder provide() {
+    return instance;
   }
 
-  @Override
-  public void start(PicoContainer container) {
-    started = true;
-  }
+  private static class NoTempFilesDuringAnalysis implements TempFolder {
 
-  @Override
-  public void stop(PicoContainer container) {
-    if (projectTempFolder != null) {
-      projectTempFolder.stop();
+    @Override
+    public File newDir() {
+      throw throwUOEFolder();
     }
-  }
 
-  @Override
-  public void dispose(PicoContainer container) {
-    // nothing to do
-  }
+    @Override
+    public File newDir(String name) {
+      throw throwUOEFolder();
+    }
 
-  @Override
-  public boolean componentHasLifecycle() {
-    return true;
-  }
+    private static UnsupportedOperationException throwUOEFolder() {
+      return new UnsupportedOperationException("Don't create temp folders during analysis");
```

Comment:
I think the file/dir name should appear in the message, e.g:
> Creation of temporary file/folder '%s' is not allowed by SonarLint

Reply:
The stacktrace will show the plugin trying to call this method, so I don't think it is very important, we will have other way to identify the root cause.